src/Microsoft.Diagnostics.Monitoring.StartupHook/Eventing/AbstractMonitorEventSource.cs

```diff
@@ -37,7 +42,15 @@ public AbstractMonitorEventSource()
 
         protected override void Dispose(bool disposing)
         {
-            _flushEventsTimer.Dispose();
+            using ManualResetEvent flushTimerFinishedHandle = new(false);
```

Comment:
> * Rather than trying to perfect our teardown path, perhaps we want to wrap our flush method in a try/catch to swallow the exception we see.

I don't think that exception is catchable. It indicates an error in the VM and shuts down immediately.

> * Intentionally leak the wait handle, or have a deferred cleanup on the wait handle.

I think if the wait completes successfully, then we can dispose it, and if it expires, then leak it. That way we don't leak in the more common happy path. I don't expect that the callback would ever take 100ms or more to complete; this was more of a mitigation to not block forever in the event that something bad did happen.